File: playwright/TC-func-REQ-SAF-001-1.ts
import { test, expect } from '@playwright/test';

test("TC-func-REQ-SAF-001-1 - Validate normal range calculation with vehicle profile available", async ({ page }) => {
  // Step 1: System reads fuel quantity 50 liters
  const fuelQuantityLocator = page.locator('[data-testid="fuel-quantity"]');
  await expect(fuelQuantityLocator).toHaveText('50 liters');

  // Step 2: System reads vehicle profile consumption 6.0 L/100km
  const vehicleConsumptionLocator = page.locator('[data-testid="vehicle-consumption"]');
  await expect(vehicleConsumptionLocator).toHaveText('6.0 L/100km');

  // Step 3: System reads last 100 km average consumption 6.0 L/100km
  const averageConsumptionLocator = page.locator('[data-testid="average-consumption"]');
  await expect(averageConsumptionLocator).toHaveText('6.0 L/100km');

  // Step 4: System calculates estimated driving range
  const calculateButtonLocator = page.locator('[data-testid="calculate-range"]');
  await calculateButtonLocator.click();

  // Step 5: System displays estimated driving range
  const estimatedRangeLocator = page.locator('[data-testid="estimated-range"]');
  await expect(estimatedRangeLocator).toHaveText('833 km');

  // Step 6: System checks low-fuel warning threshold
  const lowFuelWarningLocator = page.locator('[data-testid="low-fuel-warning"]');
  await expect(lowFuelWarningLocator).toHaveCount(0);
});

File: selenium/TC-func-REQ-SAF-001-1.java
import org.openqa.selenium.*;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.*;
import org.testng.Assert;
import org.testng.annotations.*;
import java.time.Duration;

public class TC-func-REQ-SAF-001-1_Test {
    private WebDriver driver;
    private WebDriverWait wait;

    @BeforeMethod
    public void setUp() {
        System.setProperty("webdriver.chrome.driver", "/path/to/chromedriver");
        driver = new ChromeDriver();
        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        driver.get("http://example.com"); // Replace with actual URL
    }

    @Test
    public void test_TC-func-REQ-SAF-001-1() {
        // Step 1: System reads fuel quantity 50 liters
        WebElement fuelQtyElement = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[data-testid='TODO']")));
        String fuelQtyText = fuelQtyElement.getText();
        Assert.assertEquals(fuelQtyText, "50", "Fuel quantity recorded as 50 liters");

        // Step 2: System reads vehicle profile consumption 6.0 L/100km
        WebElement vehicleConsElement = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[data-testid='TODO']")));
        String vehicleConsText = vehicleConsElement.getText();
        Assert.assertEquals(vehicleConsText, "6.0", "Vehicle profile consumption set to 6.0 L/100km");

        // Step 3: System reads last 100 km average consumption 6.0 L/100km
        WebElement avgConsElement = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[data-testid='TODO']")));
        String avgConsText = avgConsElement.getText();
        Assert.assertEquals(avgConsText, "6.0", "Average consumption set to 6.0 L/100km");

        // Step 4: System calculates estimated driving range
        WebElement estimatedRangeElement = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[data-testid='TODO']")));
        String estimatedRangeText = estimatedRangeElement.getText();
        Assert.assertTrue(estimatedRangeText.contains("833"), "Estimated range calculated as 833.33 km");

        // Step 5: System displays estimated driving range
        // Already verified in step 4

        // Step 6: System checks low-fuel warning threshold
        boolean isWarningDisplayed = driver.findElements(By.cssSelector("[data-testid='TODO']")).size() > 0 &&
                driver.findElement(By.cssSelector("[data-testid='TODO']")).isDisplayed();
        Assert.assertFalse(isWarningDisplayed, "No warning triggered because range > 80 km");
    }

    @AfterMethod
    public void tearDown() {
        if (driver != null) {
            driver.quit();
        }
    }
}